lib/http/cookie_jar/yaml_saver.rb

```diff
@@ -21,7 +21,8 @@ def save(io, jar)
 
   def load(io, jar)
     begin
-      data = YAML.load(io)
+      # determine if we have unsafe_load from psych >= 4.0.0 or otherwise use leagcy method
+      data = YAML::VERSION.method_defined?(:unsafe_load) ? YAML.safe_load(io, permitted_classes: [HTTP::Cookie, Time]) : YAML.load(io)
```

Comment:
Identifying Psych 4.x needs to change a little – here is one way to do it:

```suggestion
      data = YAML.respond_to?(:unsafe_load) ? YAML.safe_load(io, permitted_classes: [HTTP::Cookie, Time]) : YAML.load(io)
```

(Currently, this gives: `NoMethodError: undefined method 'method_defined?' for "2.0.17":String`.)

---

NB: `safe_load` has been available before Psych 4.0.0.

In order to have the same kind of behaviour as before, the [`unsafe_load`](https://github.com/ruby/psych/pull/488) method is the newly-added compatbility method.

Reply:
I read more details, discussion in https://github.com/fastlane/fastlane/pull/18803 - I can see an explicit list of classes is being prepared. Cool, I'll stand back, and let you continue this. 

Good luck!